zio-json/shared/src/main/scala/zio/json/decoder.scala

```diff
@@ -216,24 +225,22 @@ object JsonDecoder extends GeneratedTupleDecoders with DecoderLowPriority0 {
   }
   implicit val symbol: JsonDecoder[Symbol] = string.map(Symbol(_))
 
-  implicit val byte: JsonDecoder[Byte]   = number(Lexer.byte)
-  implicit val short: JsonDecoder[Short] = number(Lexer.short)
-  implicit val int: JsonDecoder[Int]     = number(Lexer.int)
-  implicit def long: JsonDecoder[Long]   = number(Lexer.long)
-  implicit val bigInteger: JsonDecoder[java.math.BigInteger] = number(
-    Lexer.bigInteger
-  )
-  implicit val float: JsonDecoder[Float]   = number(Lexer.float)
-  implicit val double: JsonDecoder[Double] = number(Lexer.double)
-  implicit val bigDecimal: JsonDecoder[java.math.BigDecimal] = number(
-    Lexer.bigDecimal
-  )
+  implicit val byte: JsonDecoder[Byte]                       = number(Lexer.byte)
+  implicit val short: JsonDecoder[Short]                     = number(Lexer.short)
+  implicit val int: JsonDecoder[Int]                         = number(Lexer.int)
+  implicit def long: JsonDecoder[Long]                       = number(Lexer.long)
+  implicit val bigInteger: JsonDecoder[java.math.BigInteger] = number(Lexer.bigInteger)
+  implicit val float: JsonDecoder[Float]                     = number(Lexer.float)
+  implicit val double: JsonDecoder[Double]                   = number(Lexer.double)
+  implicit val bigDecimal: JsonDecoder[java.math.BigDecimal] = number(Lexer.bigDecimal)
+  implicit val scalaBigDecimal: JsonDecoder[BigDecimal]      = bigDecimal.map(x => x)
```

Comment:
Unncessary map with identity function?

Reply:
Scala includes an implicit conversion from `java.math.BigDecimal` to `scala.BigDecimal` via `scala.math.javaBigDecimal2bigDecimal`, but we still have to tell the compiler how to get a `JsonDecoder[BigDecimal]` from a `JsonDecoder[java.math.BigDecimal]`. The map tells the compiler how to do the latter conversion from the former.